solr/solrj/src/java/org/apache/solr/common/cloud/PerReplicaStates.java

```diff
@@ -352,111 +349,103 @@ public int hashCode() {
    * Do not directly manipulate the per replica states as it can become difficult to debug them
    *
    */
-  public static abstract class WriteOps {
+  public static class WriteOps {
```

Comment:
I still think this class should be moved out to its own file. There is a LOT going on in here and it is deserving of a separate compilation unit.